include/erlcloud_aws.hrl

```diff
@@ -7,6 +7,8 @@
           s3_host="s3.amazonaws.com"::string(),
           s3_port=80::non_neg_integer(),
           s3_follow_redirect=false::boolean(),
+          s3_follow_redirect_count=2::non_neg_integer(),
+          s3_bucket_access_method=vhost::vhost|path,
```

Comment:
I think defaulting it to use path is pretty reasonable, but is there any actual downside to just using the path method all the time?

Also if AWS customers are getting around amazon naming conventions like this in s3, is there a risk that they are naming other resources in non compliant ways? 

If so, we should be implementing defenses like this in other areas of the API